packages/medusa/src/services/pricing.ts

```diff
@@ -220,25 +224,35 @@ class PricingService extends TransactionBaseService {
       pricingContext = await this.collectPricingContext(context)
     }
 
-    let productRates: TaxServiceRate[] = []
+    let productRates: Map<string, TaxServiceRate[]> = new Map()
+
     if (
       pricingContext.automatic_taxes &&
       pricingContext.price_selection.region_id
     ) {
+      // Here we assume that the variants belongs to the same product since the context is shared
+      const productId = variant.product_id
       productRates = await this.taxProviderService.getRegionRatesForProduct(
-        variant.product_id,
+        productId,
         {
           id: pricingContext.price_selection.region_id,
           tax_rate: pricingContext.tax_rate,
         }
       )
+      pricingContext.price_selection.tax_rates = productRates.get(productId)
     }
 
-    return await this.getProductVariantPricing_(
-      variant.id,
-      productRates,
-      pricingContext
-    )
+    return (
+      await this.getProductVariantPricing_(
+        [
+          {
+            variantId: variant.id,
+            quantity: pricingContext.price_selection.quantity,
+          },
+        ],
+        pricingContext
+      )
+    ).get(variant.id)!
```

Comment:
suggestion(non-blocking): I find this rather hard to read
```suggestion
    const singleVariantPricingMap = await this.getProductVariantPricing_(
        [
          {
            variantId: variant.id,
            quantity: pricingContext.price_selection.quantity,
          },
        ],
        pricingContext
      )
    return singleVariantPricingMap.get(variant.id)!
```